tests/functional/r/regression/regression_5030.py

```diff
@@ -0,0 +1,30 @@
+"""Regression in astroid on ClassDef inference with two test cases.
+Fixed in https://github.com/PyCQA/astroid/pull/1181"""
+
+from typing import Tuple, Type
+from typing import Dict, List, Any
+from dataclasses import dataclass, field
+
+# Example 1
+def is_type_list(f_type: Type) -> bool:
+    """just here to show the issue"""
+    return f_type == list
+
+assert not is_type_list(Tuple)
+
+# Example 2
```

Comment:
Same here. Furthermore, I would add an additional empty line just to have some separation between these two cases.